pkg/validation/spec/swagger.go

```diff
@@ -385,15 +385,15 @@ func (s SchemaOrArray) MarshalJSON() ([]byte, error) {
 	if internal.UseOptimizedJSONMarshaling {
 		return internal.DeterministicMarshal(s)
 	}
-	if len(s.Schemas) > 0 {
+	if s.Schemas != nil {
```

Comment:
this is changing the pre-existing MarshalJSON function?

Reply:
Yes, the function was NOT round-tripping with the former code either ...

Reply:
huh... surprised no fuzzing anywhere caught that

Reply:
Yeah, but if you have an empty list of schemas, this will end-up serializing `null`, which doesn't get deserialized to empty list of schemas.